### ISSUE TITLE
Support for @JsonProperty in enums

### ISSUE THREAD

**Issue by AlexandreCarlton** [victools/jsonschema-generator#66]
Support for `@JsonProperty` is offered for regular fields, but not for enumerations - currently, we have to use `@JsonValue` if we want to control the schema generation of the constants:

```java
public enum TestEnum {
    A("a"),
    B("b");

    private String value;

    TestEnum(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return value;
    }
}
```

Would it be possible to add support for `@JsonProperty` instead please? This would cut down on the amount of boilerplate needed:

```java
public enum TestEnum {
    @JsonProperty("a")
    A,
    @JsonProperty("b")
    B
}
```

Thanks!

(Also wanted to mention that this library rocks - this is the first one I've found that handles `@JsonTypeInfo` correctly!)

**Comment by CarstenWickner** [comment]
HI @AlexandreCarlton,

Thank you for your praise. 😄
I can certainly look into adding this to the `jsonschema-module-jackson` in one of the next releases.

For the time being, maybe just configure it yourself. The easiest option would be to re-use the internal `com.github.victools.jsonschema.generator.impl.module.EnumModule` class, like this:
```java
new EnumModule(value -> {
        try {
            JsonProperty annotation = value.getDeclaringClass().getDeclaredField(value.name()).getAnnotation(JsonProperty.class);
            // name() is used as fall-back if there is no @JsonProperty annotation
            return annotation == null ? value.name() : annotation.value();
        } catch (NoSuchFieldException | SecurityException ex) {
            // this should never happen
            throw new IllegalStateException(ex);
        }
    });
```
Just add this in the same way as the `JacksonModule` via `SchemaGeneratorConfigBuilder.with(Module)` and you should be good to go.

**Comment by CarstenWickner** [comment]
Hi @AlexandreCarlton,

A new `JacksonOption.FLATTENED_ENUMS_FROM_JSONPROPERTY` will be introduced in the v4.10.0 release then.

Not releasing it quite yet, since you already have some workaround and I'd rather wait a couple of days for whether there are other requests coming in. 😉

**Comment by AlexandreCarlton** [comment]
Thank you so much! :D

**Comment by CarstenWickner** [comment]
Released in [v4.10.0](https://github.com/victools/jsonschema-generator/releases/tag/v4.10.0)
